server/controllers/index.js

```diff
@@ -8,75 +8,86 @@ let currentUsers = [];
 
 /* Check if player is logged in */
 router.get('/auth', (req, res) => {
-    if (req.session.user) {
+    if (req.session.user) { /* The user session exists, authorised */
         res.status(200).json({ cookie: req.session.user });
-    } else {
-        res.sendStatus(401);
+    } else { /* The user session does not exist, unauthorised */
+        res.status(401).json({ message: "You need to log in with a username to enter the game."});
     }
 });
 
 /* Log in the user */
 router.post('/auth/login', (req, res) => {
     const user = currentUsers.find((user) => user.username === req.body.username);
-    if (!user) {
+    if (!user) { /* User does not already exist */
         req.session.user = req.body.username;            
         const user = {
             username: req.body.username
         };        
-        if (gameLogic.canJoinGame(user.username)) {
+        if (gameLogic.canJoinGame(user.username)) { /* Game has not been started yet */
             currentUsers.push(user);
             gameLogic.joinGame(user, () => {
                 res.sendStatus(200);
             });
-        } else {
-            res.status(400).json({message: "Game has already started"});
+        } else { /* Game has already began */
+            res.status(400).json({message: "Game has already started."});
         }
-    } else {
-        res.sendStatus(409);
+    } else { /* Username is taken, conflict error */
+        res.status(409).json({message: "Username already exists."});
     }    
 });
 
 /* Log out the user */
-router.post('/auth/logout', (req, res) => {
+router.get('/auth/logout', (req, res) => {
     const user = currentUsers.find((user) => user.username === req.session.user);
-    if (user) {
-        currentUsers = currentUsers.filter((otherUser) => otherUser !== user);
-        if (gameLogic.quitGame(user)) {
+    if (user) { /* User making the request exists */
```

Comment:
Don't need to check if user here, you can just use the auth service file

Reply:
true